backend/FwHeadless/Services/SyncHostedService.cs

```diff
@@ -38,6 +39,8 @@ protected override async Task ExecuteAsync(CancellationToken stoppingToken)
             }
             _projectsQueuedOrRunning.TryRemove(projectId, out var tcs);
             tcs?.TrySetResult(result);
+            // Give clients a bit more time to poll the status
+            CacheRecentSyncResult(projectId, result);
```

Comment:
There's still a race condition because the tcs is removed from the concurrent dictionary before the result is stored in the cache. We should put the result in the cache, then remove the tcs from the dictionary.